autonomy/cli/scaffold_fsm.py

```diff
@@ -0,0 +1,769 @@
+        """Initialize the utility class."""
+        self.ctx = ctx
+        self.skill_name = skill_name
+        self.dfa = dfa
+
+    @property
+    def skill_dir(self) -> Path:
+        """Get the directory to the skill."""
+        return Path(SKILLS, self.skill_name)
+
+    def do_scaffolding(self) -> None:
+        """Do the scaffolding."""
+        self._scaffold_rounds()
+        self._scaffold_behaviours()
+        self._scaffold_models()
+        self._scaffold_handlers()
+
+        # remove original 'my_model.py' file
+        (self.skill_dir / "my_model.py").unlink(missing_ok=True)
+
+        self._remove_pycache()
+        self._update_config()
+
+    def _scaffold_rounds(self) -> None:
+        """Scaffold the 'rounds.py' module."""
+        click.echo(f"Generating module {RoundFileGenerator.FILENAME}...")
+        RoundFileGenerator(self.ctx, self.skill_name, self.dfa).write_file(
+            self.skill_dir
+        )
+
+    def _scaffold_behaviours(self) -> None:
+        """Scaffold the 'behaviours.py' module."""
+        click.echo(f"Generating module {BehaviourFileGenerator.FILENAME}...")
+        BehaviourFileGenerator(self.ctx, self.skill_name, self.dfa).write_file(
+            self.skill_dir
+        )
+
+    def _scaffold_models(self) -> None:
+        """Scaffold the 'models.py' module."""
+        click.echo(f"Generating module {ModelsFileGenerator.FILENAME}...")
+        ModelsFileGenerator(self.ctx, self.skill_name, self.dfa).write_file(
+            self.skill_dir
+        )
+
+    def _scaffold_handlers(self) -> None:
+        """Scaffold the 'handlers.py' module."""
+        click.echo(f"Generating module {HandlersFileGenerator.FILENAME}...")
+        HandlersFileGenerator(self.ctx, self.skill_name, self.dfa).write_file(
+            self.skill_dir
+        )
+
+    def _update_config(self) -> None:
+        """Update the skill configuration."""
+        click.echo("Updating skill configuration...")
+        SkillConfigUpdater(self.ctx, self.skill_dir, self.dfa).update()
+
+    def _remove_pycache(self) -> None:
+        """Remove __pycache__ folders."""
+        for path in self.skill_dir.rglob("*__pycache__*"):
+            shutil.rmtree(path, ignore_errors=True)
+
+
+def _add_abstract_round_abci_if_not_present(ctx: Context) -> None:
+    """Add 'abstract_round_abci' skill if not present."""
+    if PublicId("valory", "abstract_round_abci") not in {
+        public_id.to_latest() for public_id in ctx.agent_config.skills
+    }:
+        click.echo(
+            "Skill valory/abstract_round_abci not found in agent dependencies, adding it..."
+        )
+        add_item(ctx, SKILL, PublicId("valory", "abstract_round_abci"))
+
+
+@scaffold.command()  # noqa
+@click.argument("skill_name", type=str, required=True)
+@click.option("--spec", type=click.Path(exists=True, dir_okay=False), required=True)
+@pass_ctx
+def fsm(ctx: Context, skill_name: str, spec: str) -> None:
```

Comment:
Before merging the PR, please move all the helper functions and classes to a separate module and keep only the command definition here.

Reply:
I agree, but reviewing the other modules that's not yet done either, so I suggest we do it in one go.